.golangci.yml

```diff
@@ -44,7 +44,5 @@ linters-settings:
     min-confidence: 0.9
   govet:
     check-shadowing: true
-  gocyclo:
-    min-complexity: 15
```

Comment:
I'd rather simplify the code instead of dropping this setting.

Reply:
Suggestions on simplifying `switch` are more than welcome. However, I'd prefer to have a very good reasons if we override any linter's default configuration parameters. This particular parameter we had historically, but without justification.

Reply:
The number of the `switch`'s branches can be reduced e.g. the same way it's done for `validateFlags()`.

About the parameter... The developers of gocyclo recommend the figure to be 10-20. The default is 30 which basically makes the check useless IMHO.

We've been using this value since Fremantle if not earlier without any problems so far. I remember we discussed it thouroughly, because even 15 feels a bit extreme for some people: ["This practice was adopted by the NIST Structured Testing methodology, with an observation that since McCabe's original publication, the figure of 10 had received substantial corroborating evidence, but that in some circumstances it may be appropriate to relax the restriction and permit modules with a complexity as high as 15."](https://en.wikipedia.org/wiki/Cyclomatic_complexity#Limiting_complexity_during_development).

Reply:
Can you be more specific with your idea about be similar with`validateFlags()`.
Unlike in `validateFlags()` where `case` have bunch of similarities, this `switch` had for each `case` different action.

Reply:
E.g. you could replace
```go
	case "fpgainfo":
		err = fpgaInfo(device, quiet)
	case "fmeinfo":
		err = fmeInfo(device, quiet)
	case "portinfo":
		err = portInfo(device, quiet)
```
with just
```go
	case "fpgainfo", "fmeinfo", "portinfo":
		err = fpgaInfo(device, quiet)

```

Then `listDevices()` and `portReleaseOrAssign()` could accept `cmd` instead of bools.

Reply:
so, by passing `cmd` you're suggesting to have `switch` with similar `case` statements in two places instead one? how come that makes code more "readable"?

Reply:
Well, when reading a peice of code I try to comprehend one function at a time, not two.

Reply:
I propose keeping the cyclomatic complexity check at 15 but relax it when that hits a nice and tidy switch-case. For the relaxed functions just put a //nolint:gocyclo before the function and try to make it so that there not much more than the switch in there.

Reply:
Or that.